src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -134,21 +134,38 @@ private void EnsureInit()
         {
             if (_initialized == 0)
             {
+                bool hasMicrosoftFeatureFlagSchema = false;
+
                 IConfiguration featureManagementConfigurationSection = _configuration
+                    .GetChildren()
+                    .FirstOrDefault(section =>
+                        string.Equals(
+                            section.Key,
+                            MicrosoftFeatureManagementFields.FeatureManagementSectionName,
+                            StringComparison.OrdinalIgnoreCase));
+
+                if (featureManagementConfigurationSection != null)
+                { 
+                    hasMicrosoftFeatureFlagSchema = true;
+                }
+                else
+                {
+                    featureManagementConfigurationSection = _configuration
                     .GetChildren()
                     .FirstOrDefault(section =>
                         string.Equals(
                             section.Key,
                             ConfigurationFields.FeatureManagementSectionName,
                             StringComparison.OrdinalIgnoreCase));
+                }
 
                 if (featureManagementConfigurationSection == null && RootConfigurationFallbackEnabled)
                 {
                     featureManagementConfigurationSection = _configuration;
-                }
 
```

Comment:
The check `featureManagementConfigurationSection != null` seems unnecessary since `_configuration` is unable to be null as validated in the constructor.

Reply:
Seems it should be 

```
if (featureManagementConfigurationSection == null && RootConfigurationFallbackEnabled)
{
    hasMicrosoftFeatureFlagSchema = HasMicrosoftFeatureFlagSchema(_configuration);
}
```